plantuml-maven-plugin/src/main/java/com/github/davidmoten/plantuml/plugins/SourceFileReaderExt.java

```diff
@@ -0,0 +1,24 @@
+package com.github.davidmoten.plantuml.plugins;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.List;
+
+import net.sourceforge.plantuml.BlockUml;
+import net.sourceforge.plantuml.FileFormatOption;
+import net.sourceforge.plantuml.SourceFileReader;
+import net.sourceforge.plantuml.file.SuggestedFile;
+import net.sourceforge.plantuml.preproc.Defines;
+
+public class SourceFileReaderExt extends SourceFileReader {
+
+	public SourceFileReaderExt(Defines defines, File file, File outputDirectory, List<String> config, String charset,
+			FileFormatOption fileFormatOption) throws IOException {
+		super(defines, file, outputDirectory, config, charset, fileFormatOption);
+	}
+
+	public SuggestedFile extractSuggestedFile(BlockUml blockUml) throws FileNotFoundException {
```

Comment:
pointless use of inheritance here, please get rid of this class, just call `sourceFileReader.getSuggestedFile(blockUml)` from the method `extractPreproc`.

Reply:
Ah I see that `getSuggestedFile` is a protected method so you had to use inheritance, no worries, I'll mark this as resolved.

Reply:
I am also not happy about it, but this was the quick solution to mitigate the protected method problem. If you wish I can open a PR to PlantUML with change this method visibility to public. If they accept it we can remove this class in a later release (I hope you will not mind to release with this workaround). What do you think?